src/engine/Union.cpp

```diff
@@ -362,10 +362,15 @@ std::unique_ptr<Operation> Union::cloneImpl() const {
 }
 
 // _____________________________________________________________________________
-std::shared_ptr<Operation> Union::createSortedVariant(
-    const vector<ColumnIndex>& sortOrder) const {
-  return std::make_shared<Union>(_executionContext, _subtrees.at(0),
-                                 _subtrees.at(1), sortOrder);
+PreconditionAction Union::createSortedClone(
+    const vector<ColumnIndex>& sortColumns) const {
+  if (sortColumns.empty()) {
+    return PreconditionAction::IMPLICITLY_SATISFIED;
+  }
+  auto operation = std::make_shared<Union>(_executionContext, _subtrees.at(0),
+                                           _subtrees.at(1), sortColumns);
+  return PreconditionAction{std::make_shared<QueryExecutionTree>(
+      getExecutionContext(), std::move(operation))};
```

Comment:
Isn't that exectly what `createExecutionTree<Union>` (or `makeExecutionTree`, I always forget), 
gives you (only a single function + don't repeat the executionContext argument).